crates/subspace-farmer/src/dsn/tests.rs

```diff
@@ -317,26 +308,53 @@ async fn test_dsn_sync() {
         )
     };
 
-    let syncer_multi_farming = LegacyMultiPlotsFarm::new(
-        MultiFarmingOptions {
-            base_directory: syncer_base_directory.as_ref().to_owned(),
-            farmer_protocol_info,
-            archiving_client: rpc_client.clone(),
-            farming_client: rpc_client.clone(),
-            object_mappings: object_mappings.clone(),
-            reward_address: subspace_core_primitives::PublicKey::default(),
-            bootstrap_nodes: vec![seeder_multiaddr],
-            listen_on: vec![],
-            enable_dsn_sync: false,
-            enable_dsn_archiving: false,
-            enable_farming: false,
-            relay_server_node: None,
-        },
-        syncer_max_plot_size,
-        plot_factory,
-    )
-    .await
-    .unwrap();
+    let syncer_multi_farming = loop {
+        {
+            let object_mappings = tokio::task::spawn_blocking({
+                let path = syncer_base_directory.as_ref().join("object-mappings");
+
+                move || LegacyObjectMappings::open_or_create(path)
+            })
+            .await
+            .unwrap()
+            .unwrap();
+
+            let syncer_multi_farming = LegacyMultiPlotsFarm::new(
+                MultiFarmingOptions {
+                    base_directory: syncer_base_directory.as_ref().to_owned(),
+                    farmer_protocol_info,
+                    archiving_client: rpc_client.clone(),
+                    farming_client: rpc_client.clone(),
+                    object_mappings: object_mappings.clone(),
+                    reward_address: subspace_core_primitives::PublicKey::default(),
+                    bootstrap_nodes: vec![seeder_multiaddr.clone()],
+                    listen_on: vec![],
+                    enable_dsn_sync: false,
+                    enable_dsn_archiving: false,
+                    enable_farming: false,
+                    relay_server_node: None,
+                },
+                syncer_max_plot_size,
+                plot_factory,
+            )
+            .await
+            .unwrap();
+
+            let syncer_public_key = *syncer_multi_farming.single_plot_farms()[0].public_key();
+
+            // Make sure seeder and syncer are close enough to each other
+            if bidirectional_distance(
+                &U256::from_be_bytes(seeder_public_key.into()),
+                &U256::from_be_bytes(syncer_public_key.into()),
+            ) < U256::MAX / U256::from(seeder_max_plot_size / syncer_max_plot_size)
+            {
+                break syncer_multi_farming;
+            }
```

Comment:
Hmm, I don't get why we need this property

Reply:
There are a few edge-cases described in `test_read_sequential_pieces`, this check ensures we don't hit any of them and thus read all the pieces we expected to.